### DISCUSSION TITLE
Allow for setting `result_storage_key` globally through an ENV

### DISCUSSION THREAD

**Discussion by ConstantinoSchillebeeckx** [PrefectHQ/prefect#15828] [Ideas]
https://prefect-community.slack.com/archives/CL09KU1K7/p1729791952567529

> is there a way of setting `result_storage_key` more globally, e.g. through a ENVIRONMENT var? as opposed to defining it on each task?

> I don't think there's an environment variable that sets the result_storage_key globally. The closest thing I've found is: prefect config set `PREFECT_DEFAULT_RESULT_STORAGE_BLOCK`, which sets the default storage location for your workflow, but you'd still need to set the result_storage_key at the task level.
> IMO I think it's worth exploring adding an environment variable to achieve this. 

Given that `result_storage_key` can use the prefect context, it would be great to be able to set it to a value like "{flow_run.flow_name}/{flow_run.name}/state.json" for ALL tasks at once through an ENV.

**Comment by desertaxle** [comment]
Hey @ConstantinoSchillebeeckx! I think this suggestion makes a lot of sense! Would you be willing to submit a PR to add a setting for a default result storage key? We've been reworking the Prefect settings recently so I think it'd make sense to include this setting in the [`TaskSettings`](https://github.com/PrefectHQ/prefect/blob/685b8f617de5eb14a5c75a32bbebbbf812bd34ac/src/prefect/settings/models/tasks.py#L52) model.

**Reply by ConstantinoSchillebeeckx** [reply]
sure thing, I'll take a look this week.
